src/ripple/protocol/impl/ErrorCodes.cpp

```diff
@@ -50,83 +50,71 @@ class ErrorCategory
+        add (rpcLGR_IDXS_INVALID,      "lgrIdxsInvalid",      "Ledger indexes invalid.");
+        add (rpcLGR_IDX_MALFORMED,     "lgrIdxMalformed",     "Ledger index malformed.");
+        add (rpcLGR_NOT_FOUND,         "lgrNotFound",         "Ledger not found.");
+        add (rpcLGR_NOT_VALIDATED,     "lgrNotValidated",     "Ledger not validated.");
+        add (rpcMASTER_DISABLED,       "masterDisabled",      "Master key is disabled.");
+        add (rpcNOT_ENABLED,           "notEnabled",          "Not enabled in configuration.");
+        add (rpcNOT_IMPL,              "notImpl",             "Not implemented.");
+        add (rpcNOT_READY,             "notReady",            "Not ready to handle this request.");
+        add (rpcNOT_SUPPORTED,         "notSupported",        "Operation not supported.");
+        add (rpcNO_CLOSED,             "noClosed",            "Closed ledger is unavailable.");
+        add (rpcNO_CURRENT,            "noCurrent",           "Current ledger is unavailable.");
+        add (rpcNO_EVENTS,             "noEvents",            "Current transport does not support events.");
+        add (rpcNO_NETWORK,            "noNetwork",           "Not synced to Ripple network.");
+        add (rpcNO_PERMISSION,         "noPermission",        "You don't have permission for this command.");
+        add (rpcNO_PF_REQUEST,         "noPathRequest",       "No pathfinding request in progress.");
+        add (rpcPUBLIC_MALFORMED,      "publicMalformed",     "Public key is malformed.");
+        add (rpcSIGNING_MALFORMED,     "signingMalformed",    "Signing of transaction is malformed.");
+        add (rpcSLOW_DOWN,             "slowDown",            "You are placing too much load on the server.");
+        add (rpcSRC_ACT_MALFORMED,     "srcActMalformed",     "Source account is malformed.");
+        add (rpcSRC_ACT_MISSING,       "srcActMissing",       "Source account not provided.");
+        add (rpcSRC_ACT_NOT_FOUND,     "srcActNotFound",      "Source account not found.");
+        add (rpcSRC_CUR_MALFORMED,     "srcCurMalformed",     "Source currency is malformed.");
+        add (rpcSRC_ISR_MALFORMED,     "srcIsrMalformed",     "Source issuer is malformed.");
+        add (rpcSTREAM_MALFORMED,      "malformedStream",     "Stream malformed.");
+        add (rpcTOO_BUSY,              "tooBusy",             "The server is too busy to help you now.");
+        add (rpcTXN_NOT_FOUND,         "txnNotFound",         "Transaction not found.");
+        add (rpcUNKNOWN_COMMAND,       "unknownCmd",          "Unknown method.");
+        add (rpcSENDMAX_MALFORMED,     "sendMaxMalformed",    "SendMax amount malformed.");
+
+        // Verify that the number of entries in m_map equals the number of
+        // enums that have descriptions.  That skips rpcUNKNOWN and rpcSUCCESS,
+        // which are -1 and 0 respectively.
+        assert (static_cast<int>(rpcNOT_SUPPORTED) == m_map.size());
```

Comment:
Can we add an `rpcNUM_CODES` value (I'm not picky on the name) with a comment at the end of the enum and use that here instead (we'll have to subtract 1, but I'm fine with that)?